spring-rabbit/src/test/java/org/springframework/amqp/rabbit/connection/CachingConnectionFactoryIntegrationTests.java

```diff
@@ -92,6 +94,9 @@ public void open() {
 
 	@After
 	public void close() {
+		if (!this.connectionFactory.getVirtualHost().equals("non-existent")) {
```

Comment:
Without this check, `testReceiveFromNonExistentVirtualHost()` fails - it changes the virtual host in the connection factory.